### ISSUE TITLE
custom-resources: have a way to not add statement for every invocation, I want to supply a full policydocument myself

### ISSUE THREAD

**Issue by ashald** [aws/aws-cdk#13129]
`AwsCustomResource` requires policy as an input and always creates an explicit policy, for each instance.
This is problematic for several reasons:
1. When granting access directly to the shared execution role, this is redundant
2. There is a lot of redundancy in defining a policy per each invocation which makes it very easy to hit IAM limit of 10,240 characters per all role policies

For instance, I use `AwsCustomResource` to read S3 keys' metadata.
For each key, I supply statements granting access to the S3 operations required to read the key, as well as KMS operations to use CMK used for encryption, along with several conditions to narrow down the access grant. This way I hit the limit with just 20 invocations.

Instead, I could've used a single statement for each of the above, and provided a list of key ARNs in each of them which would've made it much more compact.

### Environment

  - **CDK CLI Version  :** 1.89
  - **Framework Version:** 1.89
  - **Node.js Version:** 12
  - **OS               :** macOS
  - **Language (Version):** Java

--- 

This is :bug: Bug Report

**Comment by rix0rrr** [comment]
The AwsCustomResource allows specifying a custom policy. You can use that.

**Comment by ashald** [comment]
@rix0rrr I'm sorry, but this issue is about the fact that the custom policy is REQUIRED.
What I'm saying is that, when I use AwsCustomResource a lot in a stack and specify policy for each invocation, I hit the 10,240 chars limit.
To workaround that, I grant required permissions directly to the AwsCustomResource execution role, but because it *REQUIRES* the policy on each invocation, I need to put some fake policy there like grant `x:y` on `*` - which still takes precious chars from the 10,240 limit.

I suggest that AwsCustomResource is modified to NOT REQUIRE policy on each invocation.

**Comment by rix0rrr** [comment]
If you don't manually specify the execution role policy, and use the same policy statement on every invocation (like `PolicyStatement({ actions: ['s3:GetMetadata'], resources: ['*'] })`) they should all dedupe into the same statement.

**Comment by rix0rrr** [comment]
By the way, using a `Role.withoutPolicyUpdates()` should also work.

**Comment by ashald** [comment]
Sorry for dropping a ball on this.

So, TL;DR is that the only thing I'm asking here for - and I don't see why it shouldn't be the case - is to allow empty `policy` on `AwsCustomResource`.
As I mentioned above, the use case is such that I add a more compact policy to the execution role it uses.

Now, going back to your suggestions - I see how they make sense, but it's not something I'd prefer to do.
It's a lot of hassle to define the entire execution role for `AwsCustomResource`, so I still prefer to be be able to add policies to it as I go. This means that while `Role.withoutPolicyUpdates()` generally works, it's not really super convenient.

As for
> If you don't manually specify the execution role policy, and use the same policy statement on every invocation (like PolicyStatement({ actions: ['s3:GetMetadata'], resources: ['*'] })) they should all dedupe into the same statement.

Assuming I'm following you on this, that's not the behavior I observed. To the best of my understanding , that's because the most recent code explicitly creates a policy for each invocation, as seen in https://github.com/aws/aws-cdk/blob/v1.89.0/packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts#L362-L367
This was changed in https://github.com/aws/aws-cdk/commit/8d23f248c7496d56c352a3db4b9d4bcccbdfe1c4, and before that policies were inlined which allowed for de-duplication.

**Comment by jacobsnapp** [comment]
@rix0rrr - I have experienced the same behaviour as @ashald. A policy is added to an AwsCustomResource for each invocation even if the role was imported using `Role.from_role_arn()` with mutable=False or when using `role.without_policy_updates()`.

**Comment by om12ar** [comment]
I'm running into the same issue as @ashald. 

The suggested work arounds did not work in my case. The policy does not dedupe into a single statement, it gets added multiple times instead.
I also was not able to use `role.withoutPolicyUpdates()` because I'm creating instances of `AwsCustomResource` in multiple stack that need to share the same role. Importing the roles across stacks will not work since i can't call `withoutPolicyUpdates` on an `IRole` object.

**Comment by mattgresham** [comment]
Unfortunately this one is still a problem
